integreat_cms/static/src/js/pages/machine-translation-overlay.ts

```diff
@@ -0,0 +1,241 @@
+    document.getElementById("machine-translation-overlay-budget-usage-result").innerHTML = sum as any as string;
+    setRemainingBudget();
+};
+
+const calculateNeededBudget = (pages: Promise<object>, pageTitles: Array<string>) => {
+    pages.then((loadedPages) => {
+        let sum = 0;
+        const values = Object.values(loadedPages);
+        values.forEach((value) => {
+            if (pageTitles.includes(value.title)) {
+                if (value.hix_score >= 15) {
+                    sum += value.amount_of_words;
+                }
+            }
+        });
+        setRequiredBudget(sum);
+    });
+};
+
+const getAllPages = (selectedPages: Array<Element>) => {
+    const titles: Array<string> = [];
+    selectedPages.forEach((row) => {
+        titles.push((row.children[3].children[0] as HTMLElement).title.trim());
+    });
+    return titles;
+};
+
+const setPagesWarning = (pageTitles: Array<string>) => {
+    let sum = 0;
+    document.getElementById("machine-translation-overlay-warning").classList.add("block");
+    document.getElementById("machine-translation-overlay-warning").classList.remove("hidden");
+    const selectedPages = getAllPages(getSelectedItems());
+    document.getElementById("machine-translation-overlay-warning-optional").innerHTML = "";
+    pageTitles.forEach((title) => {
+        if (selectedPages.includes(title)) {
+            const list = document.createElement("li");
+            list.textContent = title as string;
+            list.classList.add("list-disc");
+            list.classList.add("ml-4");
+            sum += 1;
+            document.getElementById("machine-translation-overlay-warning-optional").appendChild(list);
+        }
+    });
+    document.getElementById("machine-translation-overlay-warning-number").innerHTML = sum as any as string;
+    if (document.getElementById("machine-translation-overlay-warning-optional").children.length === 0) {
+        document.getElementById("machine-translation-overlay-warning").classList.remove("block");
+        document.getElementById("machine-translation-overlay-warning").classList.add("hidden");
+    }
+};
+
+const setPages = (pageTitles: Array<string>) => {
+    document.getElementById("machine-translation-overlay-pages").innerHTML = "";
+    document.getElementById("machine-translation-overlay-pages-optional").innerHTML = "";
+    const selectedPages = getAllPages(getSelectedItems());
+    pageTitles.forEach((title, i) => {
+        if (selectedPages.includes(title)) {
+            const list = document.createElement("li");
+            list.textContent = title as string;
+            list.classList.add("list-disc");
+            list.classList.add("ml-4");
+            if (i <= 4) {
+                document.getElementById("machine-translation-overlay-pages").appendChild(list);
+            } else {
+                document.getElementById("machine-translation-overlay-pages-optional").appendChild(list);
+            }
+        }
+    });
```

Comment:
I think building HTML elements dynamically in JS is a bit more complicated than needed - instead, I'd suggest to do the calculation which pages are allowed and which aren't in Python, as well as doing the rendering process in Django templates, and let the ajax view return the rendered HTML (see 
[`render_to_string()`](https://docs.djangoproject.com/en/3.2/topics/templates/#django.template.loader.render_to_string))...
```
return JsonResponse(
    {
        "all_pages": all_pages,
        "pages_to_be_translated": render_to_string(...),
        "pages_invalid": render_to_string(...),
    }
)
```

Reply:
On second thought, I think this is fine. I mean the JS implementation works, so probably it would be a waste of time to implement the same thing in Python. Just keep in mind this option for next time 😁

Reply:
Ok, thank you. I'll write this down in my notes